### ISSUE TITLE
Bed tilt correction (experimental) : Application & instructions

### ISSUE THREAD

**Issue by NARUTOfzr** [NARUTOfzr/Neptune_3#37]
Tutorial in progress...

**Comment by abge571** [comment]
Please, I have a problem with the tilt of my bed and I have no idea how to fix it.

**Comment by NARUTOfzr** [comment]
Can you take pictures of the leveling value?




---Original---
From: ***@***.***&gt;
Date: Tue, Dec 6, 2022 09:30 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




 
Please, I have a problem with the tilt of my bed and I have no idea how to fix it.

**Comment by abge571** [comment]
> Can you take pictures of the leveling value?
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Tue, Dec 6, 2022 09:30 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37) Please, I have a problem with the tilt of my bed and I have no idea how to fix it. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;



Yes, here you have them, I add my printer it worked fine until 1 week ago, I've been with it for 2 months and I don't understand the problem, I tried updating the firmware and adjusting the eccentrics but it didn't work, it keeps printing badly on the same side


![level](https://user-images.githubusercontent.com/119713258/205790033-288ad4e1-826e-49cb-92ed-144cac0820ec.jpeg)

**Comment by NARUTOfzr** [comment]
如果您正在使用固件版本V1_1.0.5-Beta。你可以尝试在起始代码中加入"M853 L-0.2 M-0.2 "

**Comment by abge571** [comment]
> 

Yes, I use that version, but how to add the code?

**Comment by abge571** [comment]
> 如果您正在使用固件版本V1_1.0.5-Beta。你可以尝试在起始代码中加入"M853 L-0.2 M-0.2 " 解压这个文件。打印“EN3_grid test3.gcode”我已经在这个文件中加入这段代码。 [EN3_grid test3.zip](https://github.com/NARUTOfzr/Neptune_3/files/10161770/EN3_grid.test3.zip) ![image](https://user-images.githubusercontent.com/64059773/205796279-90f34680-cc2d-47dc-9404-36d391555490.png)
> 
> 在打印中观察1234的位置：如果23位置仍然被压得很紧，则尝试继续减小“L&M数值”。您可以将图片发给我看看。 ![image](https://user-images.githubusercontent.com/64059773/205795814-ee230881-d84b-4c10-a39b-83904ccd5807.png)

OK I will try it

**Comment by abge571** [comment]
> 如果您正在使用固件版本V1_1.0.5-Beta。你可以尝试在起始代码中加入"M853 L-0.2 M-0.2 " 解压这个文件。打印“EN3_grid test3.gcode”我已经在这个文件中加入这段代码。 [EN3_grid test3.zip](https://github.com/NARUTOfzr/Neptune_3/files/10161770/EN3_grid.test3.zip) ![image](https://user-images.githubusercontent.com/64059773/205796279-90f34680-cc2d-47dc-9404-36d391555490.png)
> 
> 在打印中观察1234的位置：如果23位置仍然被压得很紧，则尝试继续减小“L&M数值”。您可以将图片发给我看看。 ![image](https://user-images.githubusercontent.com/64059773/205795814-ee230881-d84b-4c10-a39b-83904ccd5807.png)

Now the part on the left prints low and the part on the right prints well, should I lower the -0.2 of L? If so, what number?

**Comment by abge571** [comment]
> @abge571 尝试L-0.3 M-0.3
> 
> 抱歉，搞混了。尝试增加L&M，L-0.15 M-0.15

Don't worry, but I've tried with those values, and it keeps failing, now the right side prints fine, but the left side prints very close to the bed, which means it doesn't print well

**Comment by abge571** [comment]
> @abge571 尝试L-0.3 M-0.3
> 
> 抱歉，搞混了。尝试增加L&M，L-0.15 M-0.15

Print the test but now it only prints the upper part well, and the lower left and right part prints very close to the bed, would I need to do another leveling of the bed?

**Comment by NARUTOfzr** [comment]
yes. It can be leveled before trying.




---Original---
From: ***@***.***&gt;
Date: Tue, Dec 6, 2022 11:52 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




  
@abge571 尝试L-0.3 M-0.3
 
抱歉，搞混了。尝试增加L&amp;M，L-0.15 M-0.15
  
Print the test but now it only prints the upper part well, and the lower left and right part prints very close to the bed, would I need to do another leveling of the bed?

**Comment by abge571** [comment]
> yes. It can be leveled before trying.

Okay so I put L-0.3 M-0.3 but level the bed first?

**Comment by NARUTOfzr** [comment]
Before leveling, the LM value will be cleared automatically. It can be leveled directly.




---Original---
From: ***@***.***&gt;
Date: Tue, Dec 6, 2022 11:59 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




  
yes. It can be leveled before trying.
  
Okay so I put L-0.3 M-0.3 but level the bed first?

**Comment by abge571** [comment]
> Before leveling, the LM value will be cleared automatically. It can be leveled directly.

Okay I just tried leveling but it still doesn't work correctly

**Comment by NARUTOfzr** [comment]
软件补偿的效果会有偏差。很抱歉，这个方法没能帮到您。


![image](https://user-images.githubusercontent.com/64059773/205807610-0e1e63fe-d529-42f3-aa17-63faae910db2.png)
根据EF的数值：尝试这个方法 #20 。

根据AC的数值，应该尝试拧松A处热床底下的螺丝，将柱子垫高0.3mm。C处的螺丝尽量拧紧。

根据BD的数值，应该尝试拧松B处热床底下的螺丝，将柱子垫高0.5mm。D处的螺丝尽量拧紧。

**Comment by NARUTOfzr** [comment]
#10 还有根据这个方法，将黄色部分的V轮放松。
![image](https://user-images.githubusercontent.com/64059773/205814155-8fd4c18c-b4b4-4af8-b334-9defa254c569.png)

以上方法为物理方法。不需要设置“M853 L M”

**Comment by abge571** [comment]
> 软件补偿的效果会有偏差。很抱歉，这个方法没能帮到您。
> 
> ![image](https://user-images.githubusercontent.com/64059773/205807610-0e1e63fe-d529-42f3-aa17-63faae910db2.png) 根据EF的数值：尝试这个方法 #20 。
> 
> 根据AC的数值，应该尝试拧松A处热床底下的螺丝，将柱子垫高0.3mm。C处的螺丝尽量拧紧。
> 
> 根据BD的数值，应该尝试拧松B处热床底下的螺丝，将柱子垫高0.5mm。D处的螺丝尽量拧紧。

Ok I'll try that, but now I tried to level the bed and test the file you gave me with M853 L-0.15 M-0.15 and it works correctly, but when I try it in a gcode file to print something for me it doesn't print well

**Comment by abge571** [comment]
> #10 还有根据这个方法，将黄色部分的V轮放松。 ![image](https://user-images.githubusercontent.com/64059773/205814155-8fd4c18c-b4b4-4af8-b334-9defa254c569.png)
> 
> 以上方法为物理方法。不需要设置“M853 L M”

Okay i'll try that

**Comment by abge571** [comment]
> > 软件补偿的效果会有偏差。很抱歉，这个方法没能帮到您。
> > ![image](https://user-images.githubusercontent.com/64059773/205807610-0e1e63fe-d529-42f3-aa17-63faae910db2.png) 根据EF的数值：尝试这个方法 #20 。
> > 根据AC的数值，应该尝试拧松A处热床底下的螺丝，将柱子垫高0.3mm。C处的螺丝尽量拧紧。
> > 根据BD的数值，应该尝试拧松B处热床底下的螺丝，将柱子垫高0.5mm。D处的螺丝尽量拧紧。
> 
> Ok I'll try that, but now I tried to level the bed and test the file you gave me with M853 L-0.15 M-0.15 and it works correctly, but when I try it in a gcode file to print something for me it doesn't print well

Nothing has worked for me, anyway, tomorrow I will try everything you told me, if not, would there be any problem in asking you again?

**Comment by NARUTOfzr** [comment]
> > > 软件补偿的效果会有偏差。很抱歉，这个方法没能帮到您。
> > > ![image](https://user-images.githubusercontent.com/64059773/205807610-0e1e63fe-d529-42f3-aa17-63faae910db2.png) 根据EF的数值：尝试这个方法 #20 。
> > > 根据AC的数值，应该尝试拧松A处热床底下的螺丝，将柱子垫高0.3mm。C处的螺丝尽量拧紧。
> > > 根据BD的数值，应该尝试拧松B处热床底下的螺丝，将柱子垫高0.5mm。D处的螺丝尽量拧紧。
> > 
> > 
> > Ok I'll try that, but now I tried to level the bed and test the file you gave me with M853 L-0.15 M-0.15 and it works correctly, but when I try it in a gcode file to print something for me it doesn't print well
> 
> Nothing has worked for me, anyway, tomorrow I will try everything you told me, if not, would there be any problem in asking you again?

Of course. I hope I can help you with your problem.

**Comment by abge571** [comment]
Give me a moment and I'll send you a photo

>

**Comment by abge571** [comment]
> > > 
> > 
> > 
> > I calibrated the arms of the printer and the bed as you told me, but the only file that prints well is the one you gave me... But then when I print something, it doesn't print well... But the leveling is still wrong
> 
> #20 If you adjust it based on this video. The leveling values of EF will be very similar. Could you take a picture to see your leveling value? And the actual state of the print model?

I have been looking at the arm and it is true that even adjusting it it continues to print badly and the values ​​are still similar to those of yesterday
![level 2](https://user-images.githubusercontent.com/119713258/205979321-2d5f3c45-b62e-4777-a4ff-44c883598f8e.jpeg)

**Comment by NARUTOfzr** [comment]
Good morning! This value is better than the previous one, indicating that the adjustment has some effects. However, there is still some deviation, which still needs to be adjusted.




---Original---
From: ***@***.***&gt;
Date: Wed, Dec 7, 2022 01:20 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




     
I calibrated the arms of the printer and the bed as you told me, but the only file that prints well is the one you gave me... But then when I print something, it doesn't print well... But the leveling is still wrong
  
#20 If you adjust it based on this video. The leveling values of EF will be very similar. Could you take a picture to see your leveling value? And the actual state of the print model?
  
I have been looking at the arm and it is true that even adjusting it it continues to print badly and the values ​​are still similar to those of yesterday

**Comment by abge571** [comment]
> 

Good morning, now what should I do? Should I keep playing the eccentrics?

**Comment by abge571** [comment]
> Good morning! This value is better than the previous one, indicating that the adjustment has some effects. However, there is still some deviation, which still needs to be adjusted.
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Wed, Dec 7, 2022 01:20 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37) I calibrated the arms of the printer and the bed as you told me, but the only file that prints well is the one you gave me... But then when I print something, it doesn't print well... But the leveling is still wrong #20 If you adjust it based on this video. The leveling values of EF will be very similar. Could you take a picture to see your leveling value? And the actual state of the print model? I have been looking at the arm and it is true that even adjusting it it continues to print badly and the values ​​are still similar to those of yesterday — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

good afternoon, I've been trying to adjust it since the other day, but there's no way, I don't know what to do if a photo is necessary, I'll send it

**Comment by NARUTOfzr** [comment]
yes. If possible, please take some pictures. Maybe you can try printing some small models.




---Original---
From: ***@***.***&gt;
Date: Sat, Dec 10, 2022 03:55 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




  
Good morning! This value is better than the previous one, indicating that the adjustment has some effects. However, there is still some deviation, which still needs to be adjusted.
 …
 ---Original--- From: @.&gt; Date: Wed, Dec 7, 2022 01:20 AM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37) I calibrated the arms of the printer and the bed as you told me, but the only file that prints well is the one you gave me... But then when I print something, it doesn't print well... But the leveling is still wrong #20 If you adjust it based on this video. The leveling values of EF will be very similar. Could you take a picture to see your leveling value? And the actual state of the print model? I have been looking at the arm and it is true that even adjusting it it continues to print badly and the values ​​are still similar to those of yesterday — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt;
  
good afternoon, I've been trying to adjust it since the other day, but there's no way, I don't know what to do if a photo is necessary, I'll send it

**Comment by abge571** [comment]
> yes. If possible, please take some pictures. Maybe you can try printing some small models.
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Sat, Dec 10, 2022 03:55 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37) Good morning! This value is better than the previous one, indicating that the adjustment has some effects. However, there is still some deviation, which still needs to be adjusted. … ---Original--- From: @.&gt; Date: Wed, Dec 7, 2022 01:20 AM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37) I calibrated the arms of the printer and the bed as you told me, but the only file that prints well is the one you gave me... But then when I print something, it doesn't print well... But the leveling is still wrong #20 If you adjust it based on this video. The leveling values of EF will be very similar. Could you take a picture to see your leveling value? And the actual state of the print model? I have been looking at the arm and it is true that even adjusting it it continues to print badly and the values ​​are still similar to those of yesterday — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt; good afternoon, I've been trying to adjust it since the other day, but there's no way, I don't know what to do if a photo is necessary, I'll send it — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

At the end of so much adjusting and looking at the printer, the files that you sent me to help me have been of great help, apparently now it prints quite well, but the only thing I have left is one last problem, which I don't know if it has been solved, and it prints well on the left and right side but I still don't know the top and bottom side, I will inform you, thank you very much

**Comment by cristian-ivascu** [comment]
What axes do M & L control? 
I've updated to 1.0.5 Beta 6x6 FW in an attempt to dial in settings further. I've made sure the bed spacers are the same length and adjusted all eccentric nuts as well.
Still, my bed seems warped in 1 corner & overall tilted. Since it is the 6x6, I pulled the ABL values using OctoPrint terminal and added the screenshot below.

I'd like to know what M & L control in which axis. The goal would be to compensate towards the back left corner

![image](https://user-images.githubusercontent.com/8312919/207396302-90b1cfbc-d7f2-4298-aba7-4f1f17a1c808.png)

**Comment by NARUTOfzr** [comment]
As shown in the figure, assuming that the X axis has a tilt similar to that shown in the figure, if the G29 (or M420 S0) is not used for automatic leveling, and the M853 L - (0~1) code is used, the Z axis will perform tilt compensation to make the nozzle always parallel to the hot bed.




---Original---
From: ***@***.***&gt;
Date: Wed, Dec 14, 2022 01:02 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




 
What axes do M &amp; L control?
 I've updated to 1.0.5 Beta 6x6 FW in an attempt to dial in settings further. I've made sure the bed spacers are the same length and adjusted all eccentric nuts as well.
 Still, my bed seems warped in 1 corner &amp; overall tilted. Since it is the 6x6, I pulled the ABL values using OctoPrint terminal and added the screenshot below.
 
I'd like to know what M &amp; L control in which axis. The goal would be to compensate towards the back left corner

**Comment by NARUTOfzr** [comment]
I have done a lot of tests on this scheme, which will have some effect, but I am not sure whether it can help you.
You can try to set<M853 L-0.15&gt;before G28.




---Original---
From: ***@***.***&gt;
Date: Wed, Dec 14, 2022 01:02 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NARUTOfzr/Neptune_3] Neptune3：Bed tilt correction (experimental) : Application &amp; instructions (Issue #37)




 
What axes do M &amp; L control?
 I've updated to 1.0.5 Beta 6x6 FW in an attempt to dial in settings further. I've made sure the bed spacers are the same length and adjusted all eccentric nuts as well.
 Still, my bed seems warped in 1 corner &amp; overall tilted. Since it is the 6x6, I pulled the ABL values using OctoPrint terminal and added the screenshot below.
 
I'd like to know what M &amp; L control in which axis. The goal would be to compensate towards the back left corner

**Comment by smilingjackal** [comment]
Wondering if I have this straight. I can add M852 code in Gcode (with the number being desired layer height) after the load current mesh command and the z-wobble type effect will stop. Is that correct?
